src/include/cart/iv.h

```diff
@@ -439,6 +439,28 @@ crt_iv_namespace_create(crt_context_t crt_ctx, crt_group_t *grp, int tree_topo,
 int
 crt_iv_namespace_id_get(crt_iv_namespace_t *ivns, uint32_t *id);
 
+/**
+ * Associate priv data with IV namespace
+ *
+ * \param[in] ivns              Incast variable namespace handle
+ * \param[in] priv              Private user data
+ *
+ * \return                      DER_SUCCESS on success, negative value on error
+ */
+int
+crt_iv_namespace_priv_set(crt_iv_namespace_t *ivns, void *priv);
```

Comment:
If a user does the following:

    crt_iv_namespace_create(..., &ivns);
    crt_iv_namespace_priv_set(ivns, user_ivns);

he/she must deal with potential IV callbacks happened between these two calls. In today's daos, this might be as simple as avoiding progressing the crt_context_t in question. That said, if user_ivns were set atomically in the first call, users would not need to consider this at all. The crt_iv_namespace_destroy_cb_t implementation would then be the natural place to release user_ivns.

Reply:
Good point. For this case I added crt_iv_namespace_create_priv() API that takes additional optional priv pointer.